### PR TITLE
feat: add OpenStack SD support in ScrapeConfig

### PR DESCRIPTION
**Pull request by slashpai**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

feat: add OpenStack SD support in ScrapeConfig

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
feat: add OpenStack SD support in ScrapeConfig
```

### CODE REVIEW

**Comment by slashpai** [comment]
docs is failing, looks temporary

`error: 		governance.md:39,62,172 (3 occurrences): "https://twitter.com/PromOperator" not accessible; status code 400: Bad Request`

File: pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go [hunk 1]
```diff
@@ -479,6 +482,83 @@ type GCESDConfig struct {
...
+	// +optional
+	AllTenants *bool `json:"allTenants,omitempty"`
+	// Refresh interval to re-read the instance list.
+	// +optional
+	RefreshInterval *v1.Duration `json:"refreshInterval,omitempty"`
+	// The port to scrape metrics from. If using the public IP address, this must
+	// instead be specified in the relabeling rule.
+	// +optional
+	Port *int `json:"port"`
+	// availability of the endpoint to connect to.
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go:553]: (nit)
```suggestion
	// Availability of the endpoint to connect to.
```

File: pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go [hunk 2]
```diff
@@ -479,6 +482,83 @@ type GCESDConfig struct {
...
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#openstack_sd_config
+// +k8s:openapi-gen=true
+type OpenStackSDConfig struct {
+	// The OpenStack role of entities that should be discovered.
+	// +kubebuilder:validation:Enum=Instance;instance;Hypervisor;hypervisor
+	// +required
+	Role string `json:"role"`
+	// The OpenStack Region.
+	// +required
+	Region string `json:"region"`
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go:495]:
```suggestion
	// +kubebuilder:validation:MinLength=1
	Region string `json:"region"`
```

**Comment by slashpai** [comment]
@simonpasquier addressed comments :)

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Pull request merged**